### PR TITLE
Int survey view with alchemer

### PR DESCRIPTION
**Pull request by cristinabaldor**
# Pull Request

## Summary & Motivation

[//]: # When merged, this pull request will add alchemer data to the int_survey_responses table and create one view for reporting of current and historical survey data.

## Self-review

- [ ] Update **due date**, **assignee**, and **priority** on our [TEAMster Asana Project](https://app.asana.com/0/1205971774138578/1205971926225838)

    _If this is a same-day request, please flag that in Slack_

- [ ] <kbd>Format</kbd> has been run on all modified files

## Troubleshooting
- [SqlFluff Rules Reference](https://docs.sqlfluff.com/en/stable/rules.html)
- [Trunk](https://teamschools.github.io/teamster/CONTRIBUTING/#trunk)
- [dbt](https://teamschools.github.io/teamster/CONTRIBUTING/#dbt-cloud_1)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206396115435493

### CODE REVIEW

File: src/dbt/kipptaf/models/surveys/intermediate/int_surveys__survey_responses.sql [hunk 1]
```diff
@@ -47,3 +52,84 @@ left join
...
+select
+    safe_cast(sr.survey_id as string) as survey_id,
+    sr.survey_title as survey_title,
+    safe_cast(sr.response_id as string) as survey_response_id,
+    coalesce(
+        sr.response_string_value,
+        sr.response_map_value,
+        sr.response_option_value,
+        sr.response_rank_value
+    ) as answer,
```
Comment (cbini) [inline src/dbt/kipptaf/models/surveys/intermediate/int_surveys__survey_responses.sql:67]: refactor to `base_alchemer__survey_results`

File: src/dbt/kipptaf/models/surveys/intermediate/int_surveys__survey_responses.sql [hunk 2]
```diff
@@ -47,3 +52,84 @@ left join
...
+    eh.report_to_mail as manager_email,
+    eh.report_to_user_principal_name as manager_user_principal_name,
+    eh.alumni_status,
+    eh.community_grew_up,
+    eh.community_professional_exp,
+    eh.level_of_education,
+    eh.primary_grade_level_taught,
+
+    timestamp(sr.response_date_started) as date_started,
+    timestamp(sr.response_date_submitted) as date_submitted,
```
Comment (cbini) [inline src/dbt/kipptaf/models/surveys/intermediate/int_surveys__survey_responses.sql:99]: refactor to `base_alchemer__survey_results`

File: src/dbt/kipptaf/models/surveys/intermediate/int_surveys__survey_responses.sql [hunk 3]
```diff
@@ -47,3 +52,84 @@ left join
...
+            )
+            is null
+        then 1
+        else 0
+    end as is_open_ended,
+from {{ ref("base_alchemer__survey_results") }} as sr
+left join
+    {{ ref("stg_reporting__terms") }} as rt
+    on rt.name = sr.survey_title
+    and date(sr.response_date_submitted) between rt.start_date and rt.end_date
```
Comment (cbini) [inline src/dbt/kipptaf/models/surveys/intermediate/int_surveys__survey_responses.sql:126]: use refactored calc

File: src/dbt/kipptaf/models/surveys/intermediate/int_surveys__survey_responses.sql [hunk 4]
```diff
@@ -47,3 +52,84 @@ left join
...
+inner join
+    {{ ref("int_surveys__response_identifiers") }} as ri
+    on sr.survey_id = ri.survey_id
+    and sr.response_id = ri.response_id
+inner join
+    {{ ref("base_people__staff_roster_history") }} as eh
+    on ri.respondent_employee_number = eh.employee_number
+    and timestamp(sr.response_date_submitted)
```
Comment (cbini) [inline src/dbt/kipptaf/models/surveys/intermediate/int_surveys__survey_responses.sql:134]: use refactored calc

**Review by cbini — CHANGES_REQUESTED** [review]

File: src/dbt/kipptaf/models/surveys/intermediate/int_surveys__survey_responses.sql (hunk 1 above)
Comment (cristinabaldor) [reply src/dbt/kipptaf/models/surveys/intermediate/int_surveys__survey_responses.sql:67]: hey, i'm not clear on what to change -- the coalesce? data types?

Comment (cbini) [reply src/dbt/kipptaf/models/surveys/intermediate/int_surveys__survey_responses.sql:67]: the coalesce, it's reused a bunch

File: src/dbt/kipptaf/models/alchemer/base/base_alchemer__survey_results.sql [hunk 5]
```diff
@@ -7,8 +7,8 @@ select
     sr.session_id as response_session_id,
     sr.contact_id as response_contact_id,
     sr.status as response_status,
-    sr.date_started as response_date_started,
-    sr.date_submitted as response_date_submitted,
+    timestamp(sr.date_started) as response_date_started,
+    timestamp(sr.date_submitted) as response_date_submitted,
```
Comment (cbini) [inline src/dbt/kipptaf/models/alchemer/base/base_alchemer__survey_results.sql:11]: can you make these new columns (i.e. `response_date_started_timestamp`) and preserve the originals?

File: src/dbt/kipptaf/models/surveys/intermediate/int_surveys__survey_responses.sql (hunk 3 above)
Comment (cbini) [reply src/dbt/kipptaf/models/surveys/intermediate/int_surveys__survey_responses.sql:103]: missed this one

File: src/dbt/kipptaf/models/surveys/intermediate/int_surveys__survey_responses.sql (hunk 4 above)
Comment (cbini) [reply src/dbt/kipptaf/models/surveys/intermediate/int_surveys__survey_responses.sql:111]: missed this one too

**Review by cbini — CHANGES_REQUESTED** [review]

**Comment by cristinabaldor** [comment]
ugh i don't even think they needed to be timestamps, they already are, i just kept the one that needs to be cast as a date to match to reporting_terms

File: src/dbt/kipptaf/models/surveys/intermediate/int_surveys__survey_responses.sql (hunk 3 above)
Comment (cbini) [reply src/dbt/kipptaf/models/surveys/intermediate/int_surveys__survey_responses.sql:103]: @cristinabaldor still need this date transformation refactored to `base_alchemer__survey_results`

Comment (cristinabaldor) [reply src/dbt/kipptaf/models/surveys/intermediate/int_surveys__survey_responses.sql]: ok one more try

**Review by cbini — APPROVED** [review]

**Pull request merged**